app/views/helpers/summary/Card7ForTransport.scala

```diff
@@ -0,0 +1,241 @@
+        classes = "borderTransport",
+        changeLink(TransportLeavingTheBorderController.displayPage, "transport.departure.transportCode.header", actionsEnabled)
+      )
+    }
+
+  private def warehouseIdentification(declaration: ExportsDeclaration, actionsEnabled: Boolean)(implicit messages: Messages): Option[SummaryListRow] =
+    declaration.locations.warehouseIdentification.map { warehouseIdentification =>
+      SummaryListRow(
+        key(
+          if (warehouseIdentification.identificationNumber.isDefined) "transport.warehouse.id"
+          else "transport.warehouse.no.label"
+        ),
+        value(warehouseIdentification.identificationNumber.getOrElse(messages("site.no"))),
+        classes = "warehouseId",
+        changeLink(WarehouseIdentificationController.displayPage, "transport.warehouse.id", actionsEnabled)
+      )
+    }
+
+  private def supervisingCustomsOffice(declaration: ExportsDeclaration, actionsEnabled: Boolean)(
+    implicit messages: Messages
+  ): Option[SummaryListRow] =
+    declaration.locations.supervisingCustomsOffice.flatMap(_.supervisingCustomsOffice.map { supervisingCustomsOffice =>
+      SummaryListRow(
+        key("transport.supervisingOffice"),
+        value(supervisingCustomsOffice),
+        classes = "supervisingOffice",
+        changeLink(SupervisingCustomsOfficeController.displayPage, "transport.supervisingOffice", actionsEnabled)
+      )
+    })
+
+  private def inlandOrBorder(declaration: ExportsDeclaration, actionsEnabled: Boolean)(implicit messages: Messages): Option[SummaryListRow] =
+    declaration.locations.inlandOrBorder.map { inlandOrBorder =>
+      SummaryListRow(
+        key("transport.inlandOrBorder"),
+        valueKey(s"declaration.summary.transport.inlandOrBorder.${inlandOrBorder.location}"),
+        classes = "inlandOrBorder",
+        changeLink(InlandOrBorderController.displayPage, "transport.inlandOrBorder", actionsEnabled)
+      )
+    }
+
+  private def inlandModeOfTransport(declaration: ExportsDeclaration, actionsEnabled: Boolean)(implicit messages: Messages): Option[SummaryListRow] =
+    inlandModeOfTransportCode(declaration).flatMap {
+      _.inlandModeOfTransportCode.map { inlandModeOfTransportCode =>
+        SummaryListRow(
+          key("transport.inlandModeOfTransport"),
+          valueKey(s"declaration.summary.transport.inlandModeOfTransport.$inlandModeOfTransportCode"),
+          classes = "modeOfTransport",
+          changeLink(InlandTransportDetailsController.displayPage, "transport.inlandModeOfTransport", actionsEnabled)
+        )
+      }
+    }
+
+  private def transportReference(transport: Transport, actionsEnabled: Boolean)(implicit messages: Messages): Option[SummaryListRow] =
+    if (transport.meansOfTransportOnDepartureType.isDefined || transport.meansOfTransportOnDepartureIDNumber.isDefined) {
+
+      val messagesForDepartureMeansOfTransport: Seq[String] =
```

Comment:
Honestly this block seems a bit complicated imo. It could be just simplified in
```
List(
  transport.meansOfTransportOnDepartureType.map(meansType => messages(s"declaration.summary.transport.departure.meansOfTransport.$meansType")),
  transport.meansOfTransportOnDepartureIDNumber.map(identity)
).flatten
```

Reply:
Agreed, ugly code, its re-used code from the section we're replacing. I'll have a look at refactoring.

(see: https://github.com/hmrc/customs-declare-exports-frontend/blame/main/app/views/declaration/summary/sections/transport_section.scala.html#L39)

Looks like its been there 4 years!